presto-main/src/main/java/io/prestosql/sql/rewrite/ShowStatsRewrite.java

```diff
@@ -123,19 +128,29 @@ protected Node visitShowStats(ShowStats node, Void context)
             if (node.getRelation() instanceof TableSubquery) {
                 Query query = ((TableSubquery) node.getRelation()).getQuery();
                 QuerySpecification specification = (QuerySpecification) query.getQueryBody();
-                Plan plan = queryExplainer.get().getLogicalPlan(session, query(specification), parameters, warningCollector);
+                Plan plan = getPlanForSpecification(specification);
```

Comment:
I would undo this change. It is simple refactor not related to the main change. Also value of this refactor is questionable. If you see a value in this, then please extract this to separate commit.

Reply:
Ok